### PR TITLE
Show MIX_XDG and HEX_HOME as sources on mix hex.config

### LINKED ISSUE #850: mix hex.config is incorrect when using MIX_XDG
Screenshot from 2021-01-13 08-00-18

the `home` config says "(default)" but it is derived from the MIX_XDG flag being set.
perhaps it should say "XDG_DATA_HOME"? Otherwise, just some way to indicate it was derived because the MIX_XDG flag was set.

### PR DESCRIPTION
**Pull request by davydog187**
Closes #850

Paired with my dearest @supersimple on this one

### CODE REVIEW

File: lib/hex/config.ex [hunk 1]
```diff
@@ -70,7 +70,7 @@ defmodule Hex.Config do
   end
 
   defp user_setting_dir!(setting) do
-    {:ok, :filename.basedir(setting, "hex", %{os: :linux})}
+    {{:env, "MIX_XDG"}, :filename.basedir(setting, "hex", %{os: :linux})}
```
Comment (ericmj) [inline lib/hex/config.ex:73]: This ignores the `XDG_*_HOME` variables, but maybe that's fine. The alternative is to not use `:filename.basedir/3` and implement it ourselves again to print the proper variable.

WDYT @wojtekmach?

**Review by ericmj — COMMENTED** [review]
Beautiful solution. Just left one comment that I would like some input on.

File: lib/hex/config.ex (hunk 1 above)
Comment (wojtekmach) [reply lib/hex/config.ex:73]: looks like it works!

```
~% XDG_CONFIG_HOME=/tmp mix hex.config | grep config_home
config_home: "/tmp/hex" (using `MIX_XDG`)
```

**Review by wojtekmach — APPROVED** [review]

File: lib/hex/config.ex (hunk 1 above)
Comment (ericmj) [reply lib/hex/config.ex:73]: But shouldn't it show the following?

```
config_home: "/tmp/hex" (using `XDG_CONFIG_HOME `)
```

Comment (wojtekmach) [reply lib/hex/config.ex:73]: that'd be option 2 from https://github.com/hexpm/hex/issues/850#issuecomment-759597741 :-)

for me personally either of these options sounds good.

Comment (davydog187) [reply lib/hex/config.ex:73]: @ericmj I believe the team decided to go this route https://github.com/hexpm/hex/issues/850#issuecomment-759597741

**Review by ericmj — APPROVED** [review]

**Pull request merged**